clustered/common/src/main/java/org/ehcache/clustered/common/PoolAllocation.java

```diff
@@ -72,6 +74,24 @@ public long getSize() {
     public String getResourceName() {
       return resourceName;
     }
+
+    @Override
+    public boolean isCompatible(PoolAllocation other) {
```

Comment:
We have a root type for doing compatibility checks: `ServerStoreCompatibility`
So either we use that type only and move all compatibility checks in it _or_ we agree that we delegate the compatibility check to the graph composing a `ServerStoreConfiguration`. But I do not understand why we would do a bit of both like is the case now.